programs/rodeos/wasm_ql_http.cpp

```diff
@@ -489,12 +492,36 @@ class http_session {
          queue_(*this) {}
 
    // Start the session
-   void run() { do_read(); }
+   void run() {
+      _timer.reset(new boost::asio::steady_timer(derived_session().stream.socket().get_executor()));
+      last_activity_timepoint = steady_clock::now();
+      start_socket_timer();
+      do_read(); 
+   }
 
  private:
    SessionType& derived_session() {
       return static_cast<SessionType&>(*this);
    }
+   
+   void start_socket_timer() 
+   {       
+      _timer->expires_after( http_config->idle_timeout_ms );
+      _timer->async_wait( [ this ]( beast::error_code ec ) {
+         if ( ec ){
+            return;
+         }
+         auto session_duration    = steady_clock::now() - last_activity_timepoint;
+         if ( session_duration <= http_config->idle_timeout_ms ){
+            start_socket_timer();
+         }
+         else{
+            ec = beast::error::timeout;
+            fail( ec, "timeout" );
+            return do_close();
+         }
+      });
+   }
```

Comment:
Above logic handles timeouts at the level of the networking session thread.  However thats not where the bulk of the work is being done.  Does the actual processing of the WasmQL query have a similar timeout (esp. wrt to JSON serialization)?  If not then closing connection does not reduce load on the rodeos node much, because it will continue to process the query from a client whose connection has been closed anyway.

Though that issue is technically beyond the scope of this PR.  Either way should be clarified in the PR description.

Reply:
Very good point, this timer is supposed to kick out idle connections, and if a connection has any activity (read/write) timer will be reset. So, it's not desirable to terminate a connection that is waiting to receive response. The minimum timeout period must be bigger than the amount of time rodeos needs to process a WasmQl query.